pkg/componentinfo/component_info.go

```diff
@@ -38,13 +38,25 @@ func (ci *ComponentInfo) KappControllerVersion() (semver.Version, error) {
 // KubernetesVersion returns the running K8s version depending on AppSpec
 // If AppSpec points to external cluster, we use that k8s version instead
 func (ci *ComponentInfo) KubernetesVersion(serviceAccountName string, specCluster *v1alpha1.AppCluster, objMeta *metav1.ObjectMeta) (semver.Version, error) {
+
+	// parses and then scrubs Pre and Build from the version
+	parseAndScrub := func(s string) (semver.Version, error) {
```

Comment:
nit: more conventionally this function would be turned into a method on ComponentInfo. i would also include "version" in the same so that it's cleat what it is doing => 

```
func (ci *ComponentInfo) parseAndScrubVersion(...
```